### PR TITLE
feat: add webview API for extensions

### PR DESCRIPTION
**Pull request by benoitf**
### What does this PR do?
- add *.d.ts definition for registering webviews
- implement API

### What issues does this PR fix or reference?

part of https://github.com/containers/podman-desktop/issues/5140

### How to test this PR?

unit tests provided

### CODE REVIEW

File: packages/extension-api/src/extension-api.d.ts [hunk 1]
```diff
@@ -1320,6 +1320,149 @@ declare module '@podman-desktop/api' {
...
+     */
+    html: string;
+
+    /**
+     * Fired when the webview content posts a message.
+     *
+     * Webview content can post strings or json serializable objects back to an extension.
+     */
+    // eslint-disable-next-line @typescript-eslint/no-explicit-any
+    readonly onDidReceiveMessage: Event<any>;
```
Comment (axel7083) [inline packages/extension-api/src/extension-api.d.ts]: Could we use unknown instead of any ?

File: packages/extension-api/src/extension-api.d.ts [hunk 2]
```diff
@@ -1320,6 +1320,149 @@ declare module '@podman-desktop/api' {
...
+
+    /**
+     * Post a message to the webview content.
+     */
+    // eslint-disable-next-line @typescript-eslint/no-explicit-any
+    postMessage(message: any): Promise<boolean>;
```
Comment (axel7083) [inline packages/extension-api/src/extension-api.d.ts]: Same remark

File: packages/main/src/plugin/api/webview-info.ts [hunk 3]
```diff
@@ -0,0 +1,29 @@
...
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ ***********************************************************************/
+
+export interface WebviewInfo {
+  id: string;
+  uuid: string;
+  viewType: string;
+  sourcePath: string;
+  icon: string | { readonly light: string; readonly dark: string } | undefined;
```
Comment (axel7083) [inline packages/main/src/plugin/api/webview-info.ts:24]: Not related to this PR. We are starting using a lot this type for icons and colors, we might want to create a type definition somewhere and reuse it

image

**Review by axel7083 — APPROVED** [review]
LGTM, Great work 👍 !

File: packages/extension-api/src/extension-api.d.ts (hunk 1 above)
Comment (benoitf) [reply packages/extension-api/src/extension-api.d.ts]: yes, implementation has changed from any to unknwown but it didn't reach the API exposure
will update

**Review by lstocchi — APPROVED** [review]
LGTM

File: packages/extension-api/src/extension-api.d.ts (hunk 1 above)
Comment (benoitf) [reply packages/extension-api/src/extension-api.d.ts]: fixed

**Comment by feloy** [comment]
Is this PR the equivalent of the `webview` branch weare working on?

When running on this branch, I can see that the extension is corrected loaded, but I cannot see an icon to access the webview.

webview-no-icon

File: packages/main/src/plugin/api/webview-info.ts (hunk 3 above)
Comment (benoitf) [reply packages/main/src/plugin/api/webview-info.ts:24]: yes

**Comment by benoitf** [comment]
@feloy you need to add also commits from https://github.com/containers/podman-desktop/pull/5593 and https://github.com/containers/podman-desktop/pull/5594

**Review by feloy — APPROVED** [review]
works great, thanks

File: packages/extension-api/src/extension-api.d.ts (hunk 2 above)
Comment (benoitf) [reply packages/extension-api/src/extension-api.d.ts]: fixed

**Comment by benoitf** [comment]
merging this PR as it has been approved by 2+ person, I'll then rebase the other 2

**Pull request merged**